### PR TITLE
Port mastodon/mastodon#18357 (Remove legacy decorators syntax) to Glitch

### PR DESCRIPTION
**Pull request by neatchee**
~~* Merge d1b057a0ac41a5756b6a416d2f290b086cb59220 - Remove legacy decorators syntax~~
~~* Merge 6e8711ff91e01212ac7c82de6349fa2b143582e2 - Add missing parenthesis~~
* Apply decorator syntax changes to app/javascript/flavours/glitch/...

### CODE REVIEW

**Comment by neatchee** [comment]
Applied, tested, and verified at neatchee/mastodon#59

**Comment by Plastikmensch** [comment]
> * Merge [d1b057a](https://github.com/glitch-soc/mastodon/commit/d1b057a0ac41a5756b6a416d2f290b086cb59220) - Remove legacy decorators syntax
> * Merge [6e8711f](https://github.com/glitch-soc/mastodon/commit/6e8711ff91e01212ac7c82de6349fa2b143582e2) - Add missing parenthesis

Those two have been merged upstream and will be included in the next upstream merge, so it's not necessary to include those.
The port itself looks okay to me, but I just skimmed it and didn't see anything immediately wrong.
Even though I'm confused about the changes of some class names, but those are also present upstream.

File: app/javascript/flavours/glitch/features/ui/components/compose_panel.jsx
Comment (Plastikmensch) [inline app/javascript/flavours/glitch/features/ui/components/compose_panel.jsx:1]: This file is missing a new line at the end

**Comment by neatchee** [comment]
> > * Merge [d1b057a](https://github.com/glitch-soc/mastodon/commit/d1b057a0ac41a5756b6a416d2f290b086cb59220) - Remove legacy decorators syntax
> > * Merge [6e8711f](https://github.com/glitch-soc/mastodon/commit/6e8711ff91e01212ac7c82de6349fa2b143582e2) - Add missing parenthesis
> 
> Those two have been merged upstream and will be included in the next upstream merge, so it's not necessary to include those. The port itself looks okay to me, but I just skimmed it and didn't see anything immediately wrong. Even though I'm confused about the changes of some class names, but those are also present upstream.

Yeah, I didn't want to potentially submit a PR that would leave anything in a broken or mismatched state; I'm happy to wait until you've merged the upstream changes and then cherry-pick just the glitch-specific work into a new PR if that's what you'd prefer. I don't trust my knowledge of React enough to assess with confidence whether it could be safely submitted on its own heh

File: app/javascript/flavours/glitch/features/compose/components/options.jsx
Comment (Plastikmensch) [inline app/javascript/flavours/glitch/features/compose/components/options.jsx:115]: Not wrong and I'm not sure if there is a reason for it, but while this is changed anyway I would prefer `(state, { name }) => (...)` to be a mapStateToProps for easier readability.

**Review by Plastikmensch — CHANGES_REQUESTED** [review]

File: app/javascript/flavours/glitch/features/compose/components/options.jsx
Comment (neatchee) [reply app/javascript/flavours/glitch/features/compose/components/options.jsx:115]: I think the upstream 'style guide' (as it lives in Eugen's head, perhaps) does single mappings in-line, but I agree that's kind of silly and makes things harder to grok

**Comment by neatchee** [comment]
This is what I get for constructing this PR across multiple text editors >.> Fixes incoming.

**Comment by Plastikmensch** [comment]
> This is what I get for constructing this PR across multiple text editors >.> Fixes incoming.

Thanks!

I'll take another thorough look when I have more energy, but looks good to me.

I see some oddities though, but I'm not sure if there is a logical difference between doing
`export default injectIntl(connect(makeMapStateToProps, mapDispatchToProps)(AccountCard));` and 
`export default connect(makeMapStateToProps, mapDispatchToProps)(injectIntl(AccountCard));`
I would personally prefer the latter as I find it easier to understand

**Comment by neatchee** [comment]
No clue. I agree that I can't think of a reason it would behave any
differently, but I certainly trust the upstream maintainers more than I
trust myself when it comes to the inner workings lol

**Comment by Plastikmensch** [comment]
> > Those two have been merged upstream and will be included in the next upstream merge, so it's not necessary to include those.
> 
> Yeah, I didn't want to potentially submit a PR that would leave anything in a broken or mismatched state; I'm happy to wait until you've merged the upstream changes and then cherry-pick just the glitch-specific work into a new PR if that's what you'd prefer. I don't trust my knowledge of React enough to assess with confidence whether it could be safely submitted on its own heh

Sorry, I completely missed replying to this.
It should be fine.
There is no need to open a new PR, you can rebase the first two commits out and force push the changes:
1. `git rebase -i <glitch-soc>/main`
2. In the text editor change "pick" to "drop" on the first two commits and save
3. `git push -f <glitch-soc>`

Where `<glitch-soc>` is the remote, in most cases `upstream`
It's worth doing a practice run (without the third step) on a copy of your branch to see everything works as expected.

File: app/javascript/flavours/glitch/components/edited_timestamp/index.jsx
Comment (Plastikmensch) [inline app/javascript/flavours/glitch/components/edited_timestamp/index.jsx:70]:
```suggestion
export default connect(null, mapDispatchToProps)(injectIntl(EditedTimestamp));
```
Missing `injectIntl`

**Comment by ClearlyClaire** [comment]
I would leave out the upstream changes themselves, and squash the commits into one porting the upstream one—like I do in the merge upstream PRs.

**Comment by Plastikmensch** [comment]
> I would leave out the upstream changes themselves, and squash the commits into one porting the upstream one—like I do in the merge upstream PRs.

Somewhat off-topic, but are you generally okay with other people porting big changes like this?
I was thinking about doing that too to ease your workload, but I was worried it would mess with your workflow or increase the load (with the review overhead) instead.

**Comment by neatchee** [comment]
> Somewhat off-topic, but are you generally okay with other people porting big changes like this? I was thinking about doing that too to ease your workload, but I was worried it would mess with your workflow or increase the load (with the review overhead) instead.

If it is the case that it impacts workflow, I would love to help document the preferred workflows for the project and give guidance to potential contributors on the best way to get their changes reviewed quickly and merged easily.

I've had two potential contributors from my instance go through me simply because I have experience in professinal dev pipelines, despite being a much worse engineer (my actual job is engineering-lite/adjacent; ref: this CR lol).

Point is, I'd be happy to spearhead an effort to take some work off your shoulders by improving the quality of the pull requests you receive through documentation :)

File: app/javascript/flavours/glitch/features/compose/components/options.jsx
Comment (ClearlyClaire) [inline app/javascript/flavours/glitch/features/compose/components/options.jsx:90]:
```suggestion
class ToggleOptionImpl extends ImmutablePureComponent {
```

File: app/javascript/flavours/glitch/features/compose/components/options.jsx
Comment (ClearlyClaire) [inline app/javascript/flavours/glitch/features/compose/components/options.jsx:119]:
```suggestion
const ToggleOption = connect(mapStateToProps)(ToggleOptionImpl);
```

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]

**Review by ClearlyClaire — APPROVED** [review]

**Comment by ClearlyClaire** [comment]
Thank you! For some reason, `/home/claire/Sources/GIT/mastodon/app/javascript/flavours/glitch/features/status/components/detailed_status.jsx` has been left out. I added it and folded everything into #2151 which I'll try to merge shortly.

**Comment by ClearlyClaire** [comment]
Closing as it was included in latest merge

**Pull request closed without merging**
